File: python/TCPDataParsing/data_parser.py
def parse_data(data):
    properties = {}
    mode = ""
    segments = data.split('/')
    for segment in segments:
        if '=' in segment:
            key, value = segment.split('=')
            if key == 'SMDD':
                current_key = value[2:]  # Remove the 'g_' prefix
            elif key == 'LD':
                properties[current_key] = value
            elif key == '>ADR':
                mode = value    
    return (mode, properties)

def parse_response(response):
    """
    Parses the received data string into a dictionary of key-value pairs.
    """
    data = {}
    parts = response.split('/')
    for part in parts:
        if 'SMDD=' in part:
            key = part.split('=')[1][2:]  # Remove the 'g_' prefix
            if key and 'LD=' in parts[parts.index(part) + 1]:
                value = parts[parts.index(part) + 1].split('=')[1]
                data[key] = int(value)
    return data

File: lib/data_parser.js
function parseData(data) {
    const properties = {};
    let mode = "";
    let currentKey = "";
    const segments = data.split('/');

    for (let segment of segments) {
        if (segment.startsWith('>ADR=')) {
            mode = segment.split('=')[1];
        } else if (segment.startsWith('SMDD=')) {
            currentKey = segment.split('=')[1].substring(2); // remove g_ or prefix
        } else if (segment.startsWith('LD=')) {
            properties[currentKey] = segment.split('=')[1];
        }
    }

    return { mode, properties };
}

function parseResponse(response) {
    const data = {};
    const parts = response.split('/');
    
    for (let i = 0; i < parts.length; i++) {
        if (parts[i].startsWith('SMDD=')) {
            const key = parts[i].split('=')[1].substring(2);
            if (parts[i+1] && parts[i+1].startsWith('LD=')) {
                const value = parts[i+1].split('=')[1];
                data[key] = parseInt(value);
            }
        }
    }

    return data;
}

module.exports = {
    parseData,
    parseResponse
};

